src/app/Library/CrudPanel/CrudFilter.php

```diff
@@ -101,6 +101,12 @@ public function wasNotApplied()
      */
     public function apply($input = null)
     {
+        // before applying filters, store the base query count (query before any filters are applied)
+        $showEntryCount = $this->crud()->getOperationSetting('showEntryCount');
+        $unfilteredQueryCount = request('unfilteredQueryCount') ?? $this->crud()->getOperationSetting('unfilteredQueryCount');
+        $unfilteredQueryCount = ! $showEntryCount ? 0 : ($unfilteredQueryCount ?? $this->crud()->getQueryCount());
+        $this->crud()->setOperationSetting('unfilteredQueryCount', $unfilteredQueryCount);
```

Comment:
This is still a bit messy:

(1) Why have the `$showEntryCount` variable in the first place?
- first, we calculate `$showEntryCount`
- then in the next line, we don't use it
- then in the third line, we do use it, but only there
- => we could have just used `$this->crud()->getOperationSetting('showEntryCount')` inline, right? That would have been cleaner.

(2) Even after the above, seems like we're doing a lot of calculation, all on top of `$unfilteredQueryCount`. So then why not simplify all this bit of logic into a function? Even if we did it in a closure, I think it would read easier:

```suggestion
        // store the base query count, before applying the filters
        $this->crud()->setOperationSetting('unfilteredQueryCount', (function () {
            if (!$this->crud()->getOperationSetting('showEntryCount')) {
                return 0;
            }

            return $this->crud()->getRequest()->input('unfilteredQueryCount') ??
                   $this->crud()->getOperationSetting('unfilteredQueryCount') ??
                   $this->crud()->getQueryCount();
        })());
```

Wouldn't you agree?

Reply:
(3) Please notice that above I've used `$this->crud()->getRequest()->input('unfilteredQueryCount')` instead of `request('unfilteredQueryCount')`. Let's make a note that we should always use the CRUD request, not the helper. That's what we do below. Just in case it's been overriden, we should use the same thing across the file.

(4) Now that we have this bit of logic separated into a closure, we can EASILY re-use it if we ever need it again, by just moving it to a method on the `CrudPanel` object.